oss-cloud-backend/services/databaseService.js

```diff
@@ -1,116 +1,63 @@
+    pr.repo,
+    pr.owner,
+    pr.number,
+    pr.link,
+    pr.title,
+    pr.dateCreated,
+    pr.status,
+    pr.author,
+  ]);
+  return mysql.transaction()
+    .query('INSERT INTO contributions (repo, owner, number, link, title, dateCreated, status, author) VALUES ?', [prs])
+    .rollback((e) => { throw e; })
+    .commit();
 };
 
-module.exports.updateContributorPullRequests = (username, pullRequests) => {
-  const params = {
-    TableName: 'contributors',
-    Key: {
-      username,
-    },
-    UpdateExpression: 'SET contributions = :prs',
-    ExpressionAttributeValues: {
-      ':prs': pullRequests,
-    },
-    ReturnValues: 'UPDATED_NEW',
-  };
-  return new Promise((resolve, reject) => {
-    docClient.update(params, (error) => {
-      if (error) {
-        reject(error);
-      }
-      resolve(true);
-    });
-  });
-};
+module.exports.getAllContributors = () => mysql.query(
+  'SELECT * FROM contributors',
+);
 
-module.exports.getAllContributors = () => {
-  const params = {
-    TableName: 'contributors',
-  };
-  return new Promise((resolve, reject) => {
-    rawClient.scan(params, (error, data) => {
-      if (error) reject(error);
-      resolve(data.Items.map((item) => attr.unwrap(item)));
-    });
-  });
+module.exports.checkUsername = async (username) => {
+  const res = await mysql
+    .query('SELECT username FROM contributors WHERE username = ?',
+      [username]);
+  return res.length !== 0;
 };
 
-module.exports.checkUsername = (username) => {
-  const params = {
-    TableName: 'contributors',
-    Key: {
-      username,
-    },
-  };
-  return new Promise((resolve, reject) => {
-    docClient.get(params, (err, data) => {
-      if (err) {
-        reject(err);
-      }
-      if (Object.keys(data).length === 0) {
-        resolve(false);
-      }
-      resolve(true);
-    });
-  });
-};
 
 // saves contributor object in database
 // params: contributor
-module.exports.addContributor = (contributor) => {
-  const params = {
-    TableName: 'contributors',
-    Item: contributor,
-  };
-  return new Promise((resolve, reject) => {
-    docClient.put(params, (err) => {
-      if (err) {
-        reject(err);
-      } else {
-        resolve();
-      }
-    });
-  });
-};
-
-// accepts a list of contributors and extracts a list of their contributions
-const extractContributions = (contributorList) => {
-  let contributions = [];
-
-  contributorList.forEach((contributor) => {
-    contributions = contributions.concat(contributor.contributions);
-  });
-  return contributions;
-};
-
-module.exports.getAllContributions = () => {
-  const params = {
-    TableName: 'contributors',
-  };
-  return new Promise((resolve, reject) => {
-    rawClient.scan(params, (error, data) => {
-      if (error) reject(error);
-      resolve(extractContributions(data.Items.map((item) => attr.unwrap(item))));
-    });
-  });
-};
+module.exports.addContributor = (contributor) => mysql.transaction()

```

Comment:
Transaction is not needed here since only one query is defined.

Reply:
Transaction removed.